.github/workflows/build.yml

```
@@ -185,3 +185,62 @@ jobs:
     - run: make clang-debug
     - name: tests (wasm2c tests excluding memory64)
       run: ./test/run-tests.py wasm2c --exclude-dir memory64
+
+  build-arch:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        arch: [s390x]
+    services: # still faster on debian...
+      distcc:
+        image: debian:latest
+        options: --health-cmd distccmon-text --health-interval 5s --health-start-period 5m debian:latest bash -c "apt-get update && apt-get install -y g++-s390x-linux-gnu distcc && distccd --daemon --no-detach"
+        ports:
+          - 3632:3632
+    steps:
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.x'
+    - uses: actions/checkout@v1
+      with:
+        submodules: true
+    - name: Set up QEMU
+      if: ${{ matrix.arch != 'i386' }}
+      uses: docker/setup-qemu-action@v2
+      with:
+        platforms: ${{matrix.arch}}
+        image: "tonistiigi/binfmt:master"
+    - name: install ninja
+      run: sudo apt-get install ninja-build
+    - name: install the toolchain
+      run: sudo apt-get install g++-${{matrix.arch}}-linux-gnu
+    - name: install distcc
+      run: sudo apt-get install distcc
+    - name: mkdir distcc symlinks
+      run: sudo mkdir -p /opt/bin/distcc_symlinks
+    - name: distcc symlink
+      run: sudo ln -s /usr/bin/distcc /opt/bin/distcc_symlinks/${{matrix.arch}}-linux-gnu-gcc # only CC is needed
+    - name: mkdir
+      run: mkdir -p out
+    - name: cmake
+      run: cmake -DCMAKE_TOOLCHAIN_FILE=../scripts/TC-${{matrix.arch}}.cmake .. -G Ninja -DWITH_WASI=ON -DWERROR=OFF -Werror=dev -Wno-deprecated
+      working-directory: out
```

This can be simplified a little:
```suggestion
    - name: cmake
      run: cmake -G Ninja -S . -B out -DCMAKE_TOOLCHAIN_FILE=./scripts/TC-${{matrix.arch}}.cmake -DWITH_WASI=ON -DWERROR=OFF -Werror=dev -Wno-deprecated
```

---

we would rather keep this consistent with the `build` job, aside from the toolchain file (and, currently, the `-DWERROR=OFF`).

---

This isn't consistent everywhere in this file, though... look at line 161 for instance. I'd sooner move the file towards having fewer unnecessary steps (which just add noise to logs)

---

hmm... @sbc100 thoughts?

---

Maybe make separate PR to convert to using `-S + -B` everywhere before we lang this change?

---

The `-S` and `-B` flags were introduced in CMake 3.13 while the minimum version for wabt is CMake 3.16 and has been for a while. No hazard with using them here.

---

hmm even the places that use `-S + -B` aren't consistent about order. would you prefer `-S . -B out -G Ninja` or `-G Ninja -S . -B out`?

---

No opinion there..